Automattic/jetpack
# Blocks: Move UpgradeNudge into components/

#### Changes proposed in this Pull Request:

Move the UpgradeNudge component into the `extensions/shared/components/` subdirectory. I think it makes sense to move our re-usable components there so that `extensions/shared/` itself is a bit less cluttered.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

Moves files around :slightly_smiling_face: 

#### Testing instructions:

Verify that Jetpack still builds, and that the `UpgradeNudge` still works.

- Switch to this branch locally, build Jetpack (`yarn build`), and start Docker (`yarn docker:up`)
- Make sure you're connected to WP.com, and on a free plan
- Start a new post, and insert the 'Simple Payments' block
- Select the block, and enter some information into its text fields.
- Click the nudge's "Upgrade" CTA button
- Pay for the premium plan
- Verify that you're redirected back to where you came from (the wp-admin block editor) (without any intermediate redirects on WP.com)

#### Proposed changelog entry for your changes:

(Doesn't need a changelog entry I think?)
